docs/rakugo_singleton.md

```diff
@@ -51,71 +49,136 @@ func _process(delta):
+```
+
+Only methods [parse_script] or [parse_and_execute_script], use a file_path. For others and signals is a script_name.
+
 # References
 ## Signals
-### say
+### sg_say
 
-params: (character:Dictionary, text:String)
+args: (character:Dictionary, text:String)
 
 Send when a [Say] instruction is executed then Rakugo waiting call of [do_step]
 
 ```gd
 func _ready():
-  Rakugo.connect("say", self, "_on_say")
+  Rakugo.sg_say.connect(_on_say)
   
 func _on_say(character:Dictionary, text:String):
   prints("say", character.get("name", "null"), text)
 ```
 
-### step
+### sg_step
 
 Send after a [Say] instruction is executed.
 
-### ask
+```gd
+func _ready():
+  Rakugo.sg_step.connect(_on_step)
+
+func _on_step():
+  prints("Press 'Enter' to continue...")
+```
+
+### sg_ask
 
-params: (character:Dictionary, question:String, default_answer:String)
+args: (character:Dictionary, question:String, default_answer:String)
 
 Send when a [Ask] instruction is executed then Rakugo waiting call of [ask_return].
 
 ```gd
 func _ready():
-  Rakugo.connect("ask", self, "_on_ask")
+  Rakugo.sg_ask.connect(_on_ask)
   
 func _on_ask(character:Dictionary, question:String, default_answer:String):
   prints("ask", character.get("name", "null"), question, default_answer)
 ```
 
-### menu
+### sg_menu
 
-params: (choices:Array)
+args: (choices:Array)
 
 Send when a [Menu] instruction is executed then Rakugo waiting call of [menu_return].
 
 ```gd
 func _ready():
-  Rakugo.connect("menu", self, "_on_menu")
+  Rakugo.sg_menu.connect(_on_menu)
   
-func _on_menu(choices:Array):
+func _on_menu(choices:PackedStringArray):
   prints("menu", choices)
 ```
 
-### execute_script_finished
+### sg_execute_script_start
 
 args: (script_name:String)
 
-Send when a script execution is finished.
+Send when the script with this `script_name` execution is started.
+
+```gd
+func _ready():
+  Rakugo.sg_execute_script_start.connect(_on_execute_script_start)
+  
+func _on_execute_script_start(script_name:String):
+  prints("execute_script_start", script_name)
+```
+
+### sg_execute_script_finished
+
+args: (script_name:String, error_str:String)
+
+Send when the script with this `script_name` execution is finished.
+
+If execution fail `error_str` contains error message, instead is empty.
 
 ```gd
 func _ready():
-  Rakugo.connect("execute_script_finished", self, "_on_execute_script_finished")
+  Rakugo.sg_execute_script_finished.connect(_on_execute_script_finished)
   
-func _on_execute_script_finished(script_name:String):
+func _on_execute_script_finished(script_name:String, error_str:String):
   prints("execute_script_finished", script_name)
 ```
 
-### parser_unhandled_regex
+### sg_custom_regex
+
+args: (key:String, result:RegExMatch)
+
+Send when a custom instruction is executed.
+
+You can add one with [add_custom_regex].
+
+```gd
+func _ready():
+  Rakugo.sg_custom_regex.connect(_on_custom_regex)
+
+func _on_custom_regex(key:String, result:RegExMatch):
```

Comment:
Good, you renamed `parser_unhandled_regex` to `custom_regex`.
But for me this to little - there should be links to what regex is and/or regex in godot docs as it:
- https://regexr.com/
- https://docs.godotengine.org/en/stable/classes/class_regex.html

And also this example is too simple it doesn't show that you can get args from it, 
so I would add example similar to this one:
https://github.com/rakugoteam/Rakugo-Dialogue-System/issues/196#issuecomment-1425279964